src/go/k8s/api/redpanda/v1alpha1/redpanda_conversion.go

```diff
@@ -0,0 +1,228 @@
+				CloudStorageAPIEndpointPort:             cfg.CloudStorageAPIEndpointPort,
+				CloudStorageBucket:                      cfg.CloudStorageBucket,
+				CloudStorageAzureContainer:              cfg.CloudStorageAzureContainer,
+				CloudStorageAzureManagedIdentityID:      cfg.CloudStorageAzureManagedIdentityID,
+				CloudStorageAzureStorageAccount:         cfg.CloudStorageAzureStorageAccount,
+				CloudStorageAzureSharedKey:              cfg.CloudStorageAzureSharedKey,
+				CloudStorageAzureADLSEndpoint:           cfg.CloudStorageAzureADLSEndpoint,
+				CloudStorageAzureADLSPort:               cfg.CloudStorageAzureADLSPort,
+				CloudStorageCacheCheckInterval:          cfg.CloudStorageCacheCheckInterval,
+				CloudStorageCacheDirectory:              cfg.CloudStorageCacheDirectory,
+				CloudStorageCacheSize:                   cfg.CloudStorageCacheSize,
+				CloudStorageCredentialsSource:           cfg.CloudStorageCredentialsSource,
+				CloudStorageDisableTLS:                  cfg.CloudStorageDisableTLS,
+				CloudStorageEnableRemoteRead:            cfg.CloudStorageEnableRemoteRead,
+				CloudStorageEnableRemoteWrite:           cfg.CloudStorageEnableRemoteWrite,
+				CloudStorageInitialBackoffMs:            cfg.CloudStorageInitialBackoffMs,
+				CloudStorageManifestUploadTimeoutMs:     cfg.CloudStorageManifestUploadTimeoutMs,
+				CloudStorageMaxConnectionIdleTimeMs:     cfg.CloudStorageMaxConnectionIdleTimeMs,
+				CloudStorageMaxConnections:              cfg.CloudStorageMaxConnections,
+				CloudStorageReconciliationIntervalMs:    cfg.CloudStorageReconciliationIntervalMs,
+				CloudStorageRegion:                      cfg.CloudStorageRegion,
+				CloudStorageSegmentMaxUploadIntervalSec: cfg.CloudStorageSegmentMaxUploadIntervalSec,
+				CloudStorageSegmentUploadTimeoutMs:      cfg.CloudStorageSegmentUploadTimeoutMs,
+				CloudStorageTrustFile:                   cfg.CloudStorageTrustFile,
+				CloudStorageUploadCtrlDCoeff:            cfg.CloudStorageUploadCtrlDCoeff,
+				CloudStorageUploadCtrlMaxShares:         cfg.CloudStorageUploadCtrlMaxShares,
+				CloudStorageUploadCtrlMinShares:         cfg.CloudStorageUploadCtrlMinShares,
+				CloudStorageUploadCtrlPCoeff:            cfg.CloudStorageUploadCtrlPCoeff,
+				CloudStorageUploadCtrlUpdateIntervalMs:  cfg.CloudStorageUploadCtrlUpdateIntervalMs,
+			}
+			t := s.Storage.Tiered
+			tiered := &v1alpha2.Tiered{
+				MountType:                     t.MountType,
+				HostPath:                      t.HostPath,
+				TieredStoragePersistentVolume: t.TieredStoragePersistentVolume,
+				Config:                        c,
+				CredentialsSecretRef:          t.CredentialsSecretRef,
+			}
+			d.Storage = &v1alpha2.Storage{
+				HostPath:         s.Storage.HostPath,
+				PersistentVolume: s.Storage.PersistentVolume,
+				Tiered:           tiered,
+			}
+		}
+	}
+	return nil
+}
+
+// ConvertFrom converts from the Hub version (v1) to this version.
+func (dst *Redpanda) ConvertFrom(srcRaw conversion.Hub) error {
```

Comment:
Same note here.

Reply:
Ok I will try to marshal and unmarshal it.